photon-api/src/main/scala/com/linkedin/photon/ml/model/RandomEffectModelInProjectedSpace.scala

```diff
@@ -76,17 +96,21 @@ protected[ml] class RandomEffectModelInProjectedSpace(
   }
 
   /**
-   * Convert the projected space model into a random effect model
+   * Convert the projected space model into a [[RandomEffectModel]].
    *
-   * @return The random effect model
+   * @return A [[RandomEffectModel]]
    */
   def toRandomEffectModel: RandomEffectModel =
-    new RandomEffectModel(modelsInProjectedSpaceRDD, randomEffectType, featureShardId)
+    new RandomEffectModel(modelsInProjectedSpaceRDD, randomEffectType, featureShardId) {
+      // TODO: The model types don't necessarily match, but checking each time is slow so copy the type for now
+      override val modelType: TaskType = this.modelType
+    }
 
   /**
-   * Update the random effect model in projected space with new underlying models per individual
+   * Update the random effect model in projected space with new underlying models (per individual).
    *
-   * @param updatedModelsRDDInProjectedSpace The new models with coefficients in projected space
+   * @param updatedModelsRDDInProjectedSpace The new underlying models with coefficients in projected space, one per
```

Comment:
"underlying"? I've tried to use "sub-model" - but not sure what term we want to use. We should pick one and be consistent. Also not sure if your "underlying" is my "sub-". Either way, just need to have consistency, IMHO.